dtfs-central-api/src/v1/controllers/portal/gef-deal/add-min-activities.controller.js

```diff
@@ -15,18 +15,19 @@ const { PORTAL_ACTIVITY_LABEL, PORTAL_ACTIVITY_TYPE } = require('../../../../con
  * canResubmitIssuedFacilities - changes flags to false
  * @param {Object} facilities
  */
-const updateChangedToIssued = async (facilities) => {
-  facilities.forEach(async (facility) => {
-    const { _id, canResubmitIssuedFacilities } = facility;
-
-    if (canResubmitIssuedFacilities) {
-      const update = {
-        canResubmitIssuedFacilities: false,
-      };
-
-      await updateFacility(_id, update);
-    }
-  });
+const updateChangedToIssued = async ({ facilities, auditDetails }) => {
+  await Promise.all(
+    facilities.map((facility) => {
+      const { _id: facilityId, canResubmitIssuedFacilities } = facility;
+      if (canResubmitIssuedFacilities) {
+        const facilityUpdate = {
+          canResubmitIssuedFacilities: false,
+        };
+        updateFacility({ facilityId, facilityUpdate, auditDetails });
+      }
+      return Promise.resolve();
```

Comment:
I'm not a fan of this but can't think of a good alternative - should we be throwing an error if `!canResubmitIssuedFacilities`? Appreciate this is a big refactor but getting code ick here